### PR TITLE
Property AutoSync

### PR DESCRIPTION
**Pull request by garrettluskey**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behaviour?

No AutoSync

## What is the new behaviour?

This adds the functionality of automatically syncing a property between the server and clients

## Other information

### CODE REVIEW

File: source/Common/Serialization/SerializableTypeMapper.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+﻿using Common.Logging;
+using ProtoBuf;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Common.Serialization;
+
```
Comment (samuelzurowski) [inline source/Common/Serialization/SerializableTypeMapper.cs:10]: please add comment

File: source/Common/Serialization/SerializableTypeMapper.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+public interface ISerializableTypeMapper
+{
+    void AddTypes(IEnumerable<Type> types);
+    bool TryGetId(Type type, out int id);
+    bool TryGetType(int id, out Type type);
+}
+
```
Comment (samuelzurowski) [inline source/Common/Serialization/SerializableTypeMapper.cs:41]: please add comment

File: source/E2E.Tests/Services/Heroes/HeroSetNameAuto.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Localization;
+using Xunit.Abstractions;
+using static TaleWorlds.CampaignSystem.GameMenus.GameMenuEventHandler;
+
+namespace E2E.Tests.Services.Heroes;
```
Comment (samuelzurowski) [inline source/E2E.Tests/Services/Heroes/HeroSetNameAuto.cs:16]: comment por favor

File: source/GameInterface.Tests/AutoSync/DataClassCreatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+using GameInterface.Utils.AutoSync.Dynamic;
+using HarmonyLib;
+using ProtoBuf;
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+using Xunit;
+
+namespace GameInterface.Tests.AutoSync;
+public class DataClassCreatorTests
```
Comment (samuelzurowski) [inline source/GameInterface.Tests/AutoSync/DataClassCreatorTests.cs:14]: comment por favor

File: source/GameInterface.Tests/AutoSync/EventGeneratorTests.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace GameInterface.Tests.AutoSync;
+public class EventGeneratorTests
```
Comment (samuelzurowski) [inline source/GameInterface.Tests/AutoSync/EventGeneratorTests.cs:12]: comment por favor

File: source/GameInterface.Tests/AutoSync/HandlerGeneratorTests.cs [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+using GameInterface.Services.Registry;
+using GameInterface.Utils.AutoSync.Dynamic;
+using GameInterface.Utils.AutoSync.Template;
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+using Xunit;
+
+namespace GameInterface.Tests.AutoSync;
+public class HandlerGeneratorTests
```
Comment (samuelzurowski) [inline source/GameInterface.Tests/AutoSync/HandlerGeneratorTests.cs:19]: comment por favor

File: source/GameInterface.Tests/AutoSync/HarmonyMethodBuilderTests.cs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using Xunit;
+
+namespace GameInterface.Tests.AutoSync;
+public class HarmonyMethodBuilderTests
```
Comment (samuelzurowski) [inline source/GameInterface.Tests/AutoSync/HarmonyMethodBuilderTests.cs:14]: comment por favor

File: source/GameInterface/Utils/AutoSync/AutoSync.cs [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+        var results = propertySync.SyncProperty(property, stringIdGetter);
+
+        disposables.Add(propertySync);
+
+        return results;
+    }
+}
+
+public class PropertySync : IDisposable
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/AutoSync.cs:71]: comment

File: source/GameInterface/Utils/AutoSync/Dynamic/HarmonyPatchGenerator.cs [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+/// Prefix function generator for property sync
+/// </summary>
+public class HarmonyPatchGenerator
+{
+    private static readonly ILogger Logger = LogManager.GetLogger<HarmonyPatchGenerator>();
+    private readonly TypeBuilder autoPatchType;
+    private readonly Type dataClassType;
+    private readonly Type eventType;
+
+    public HarmonyPatchGenerator(ModuleBuilder moduleBuilder, PropertyInfo propertyInfo, Type dataClassType, Type eventClassType)
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/HarmonyPatchGenerator.cs:30]: please comment

File: source/pdb/pingme.txt
Comment (samuelzurowski) [inline source/pdb/pingme.txt:1]: whats this?

File: source/GameInterface/Services/MobileParties/Patches/PartyArmyPatches.cs [hunk 10]
```diff
@@ -54,3 +56,15 @@ internal static void OverrideSetArmy(MobileParty mobileParty, Army army)
         });
     }
 }
+
+[HarmonyPatch]
+class IDK
```
Comment (samuelzurowski) [inline source/GameInterface/Services/MobileParties/Patches/PartyArmyPatches.cs]: is this just to test it lol?

File: source/GameInterface/Utils/AutoSync/Template/AutoSyncHandlerTemplate.cs
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Template/AutoSyncHandlerTemplate.cs:1]: May want to add documentation to the wiki on how to use?

File: source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+            typeof(object)
+        );
+
+        // Add IAutoSyncData<dataType> interface implementation
+        typeBuilder.AddInterfaceImplementation(typeof(IAutoSyncData<>).MakeGenericType(dataType));
+
+        // Add ProtoContract attribute to the new data class
+        var attributeBuilder = new CustomAttributeBuilder(
+            typeof(ProtoContractAttribute).GetConstructor(Type.EmptyTypes),
+            new object[0],
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs]:
```suggestion
            new object[]{},
```

Comment (samuelzurowski) [reply source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs]: technically both same but makes it more clear

File: source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    private FieldBuilder CreateSerializableProperty(TypeBuilder typeBuilder, string propertyName, Type propertyType)
+    {
+        var customAttributeBuilder = new CustomAttributeBuilder(
+            typeof(ProtoMemberAttribute).GetConstructor(new Type[] { typeof(int) }),
+            new object[] { protoMemberCounter++ });
+
+        // Define a private field for the property
+        FieldBuilder fieldBuilder = typeBuilder.DefineField("_" + propertyName, propertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs]:
```suggestion
        FieldBuilder fieldBuilder = typeBuilder.DefineField($"_{propertyName}", propertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
```

File: source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+        fieldBuilder.SetCustomAttribute(customAttributeBuilder);
+
+        // Define the getter method
+        MethodBuilder getMethodBuilder = typeBuilder.DefineMethod("get_" + propertyName,
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs]:
```suggestion
        MethodBuilder getMethodBuilder = typeBuilder.DefineMethod($"get_{propertyName}",
```

File: source/GameInterface.Tests/AutoSync/EventGeneratorTests.cs [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+        var dataClassCreator = new DataClassGenerator(moduleBuilder);
+        var eventClassCreator = new EventMessageGenerator();
+
+        var testIntProperty = AccessTools.Property(typeof(EventGeneratorTests), nameof(TestInt));
+
+        // Act
+        var dataType = dataClassCreator.GenerateClass(testIntProperty.PropertyType, testIntProperty.Name);
+        var eventType = eventClassCreator.GenerateEvent(moduleBuilder, testIntProperty);
+
```
Comment (samuelzurowski) [inline source/GameInterface.Tests/AutoSync/EventGeneratorTests.cs:38]: c# question why do we need to tell the compiler "trust me its not null"?

File: source/GameInterface.Tests/AutoSync/HandlerGeneratorTests.cs [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+        ILogger logger = null;
+        var handlerInstance = (IHandler)Activator.CreateInstance(handlerType, new object[] { messageBroker, objectManager, network, logger, syncedProperty });
+
+        // Assert
+        Assert.NotNull(handlerInstance);
+
+        var dataInstance = Activator.CreateInstance(dataClassType, new object[] { testObj.StringId, testObj.SomeValue })!;
+        var eventInstance = Activator.CreateInstance(eventType, new object[] { dataInstance })!;
+
+        typeof(MessageBroker).GetMethod(nameof(MessageBroker.Publish)).MakeGenericMethod(eventType)
```
Comment (samuelzurowski) [inline source/GameInterface.Tests/AutoSync/HandlerGeneratorTests.cs:69]: why is this requiring null parameter? Shouldnt it just be ```c# new object[] { eventInstance }```

File: source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+        this.moduleBuilder = moduleBuilder;
+    }
+
+    public Type GenerateClass(Type dataType, string name)
+    {
+        // Creates type builder for the new data class
+        TypeBuilder typeBuilder = moduleBuilder.DefineType(
+            $"{name}Data",
+            TypeAttributes.Public | TypeAttributes.Sealed | TypeAttributes.AnsiClass | TypeAttributes.AutoLayout | TypeAttributes.BeforeFieldInit,
+            typeof(object)
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs]: Rather then being a typeof(object). For tracking wouldn't something like this be useful? so you can verify the field. unless its just string, int, etc may be pointless.

```c#
public class CustomDataTypeGenerator : Object { }

...

        TypeBuilder typeBuilder = moduleBuilder.DefineType(
            $"{name}Data",
            TypeAttributes.Public | TypeAttributes.Sealed | TypeAttributes.AnsiClass | TypeAttributes.AutoLayout | TypeAttributes.BeforeFieldInit,
            typeof(CustomDataTypeGenerator)
        );
```

File: source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+        typeBuilder.DefineMethodOverride(getHashCodeMethodBuilder, typeof(object).GetMethod("GetHashCode"));
+    }
+
+    private FieldBuilder CreateProperty(TypeBuilder typeBuilder, string propertyName, Type propertyType)
+    {
+        // Define a private field for the property
+        FieldBuilder fieldBuilder = typeBuilder.DefineField("_" + propertyName, propertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
+
+        // Define the getter method
+        MethodBuilder getMethodBuilder = typeBuilder.DefineMethod("get_" + propertyName,
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs]:
```suggestion
        MethodBuilder getMethodBuilder = typeBuilder.DefineMethod($"get_{propertyName}",
```

File: source/GameInterface/Utils/AutoSync/Template/IAutoSyncData.cs [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+namespace GameInterface.Utils.AutoSync.Template;
+
+/// <summary>
+/// Interface for auto sync data
+/// </summary>
+/// <typeparam name="DataType">Type of data to sync, this will be either a field or property type. This type must also be serializable by protobuf</typeparam>
+public interface IAutoSyncData<DataType>
+{
+    string StringId { get; }
+    DataType Value { get; }
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Template/IAutoSyncData.cs:13]: How is this data's type validated over the network?

File: source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+                                        propertyType, null);
+
+        // Generate IL for the getter method
+        ILGenerator ilGenerator = getMethodBuilder.GetILGenerator();
+        ilGenerator.Emit(OpCodes.Ldarg_0);                 // Load "this" onto the stack
+        ilGenerator.Emit(OpCodes.Ldfld, fieldBuilder);     // Load the field onto the stack
+        ilGenerator.Emit(OpCodes.Ret);                     // Return the field value
+
+        // Define the property
+        PropertyBuilder propertyBuilder = typeBuilder.DefineProperty(propertyName, PropertyAttributes.HasDefault, propertyType, null);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs]:
```suggestion
        PropertyBuilder propertyBuilder = typeBuilder.DefineProperty(propertyName, PropertyAttributes.HasDefault, propertyType, Type.EmptyTypes);
```

Comment (samuelzurowski) [reply source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs]: just to be explicit

File: source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+        ilGenerator.Emit(OpCodes.Ret); // Return the hash code
+
+        // Override GetHashCode method
+        typeBuilder.DefineMethodOverride(getHashCodeMethodBuilder, typeof(object).GetMethod("GetHashCode"));
+    }
+
+    private FieldBuilder CreateProperty(TypeBuilder typeBuilder, string propertyName, Type propertyType)
+    {
+        // Define a private field for the property
+        FieldBuilder fieldBuilder = typeBuilder.DefineField("_" + propertyName, propertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs]:
```suggestion
        FieldBuilder fieldBuilder = typeBuilder.DefineField($"_{propertyName}", propertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
```

File: source/GameInterface/Utils/AutoSync/Dynamic/EventMessageGenerator.cs [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+                                        propertyType, null);
+
+        // Generate IL for the getter method
+        ILGenerator ilGenerator = getMethodBuilder.GetILGenerator();
+        ilGenerator.Emit(OpCodes.Ldarg_0);                 // Load "this" onto the stack
+        ilGenerator.Emit(OpCodes.Ldfld, fieldBuilder);     // Load the field onto the stack
+        ilGenerator.Emit(OpCodes.Ret);                     // Return the field value
+
+        // Define the property
+        PropertyBuilder propertyBuilder = typeBuilder.DefineProperty(propertyName, PropertyAttributes.HasDefault, propertyType, null);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/EventMessageGenerator.cs:113]:
```suggestion
        PropertyBuilder propertyBuilder = typeBuilder.DefineProperty(propertyName, PropertyAttributes.HasDefault, propertyType, Type.EmptyTypes);
```

File: source/GameInterface/Utils/AutoSync/Dynamic/EventMessageGenerator.cs [hunk 22]
```diff
@@ -0,0 +1,120 @@
...
+    private FieldBuilder CreateProperty(TypeBuilder typeBuilder, string propertyName, Type propertyType)
+    {
+        // Define a private field for the property
+        FieldBuilder fieldBuilder = typeBuilder.DefineField("_" + propertyName, propertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
+
+        // Define the getter method
+        MethodBuilder getMethodBuilder = typeBuilder.DefineMethod("get_" + propertyName,
+                                        MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.SpecialName |
+                                        MethodAttributes.NewSlot | MethodAttributes.Virtual | MethodAttributes.Final,
+                                        propertyType, null);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/EventMessageGenerator.cs:104]:
```suggestion
                                        propertyType, Type.EmptyTypes);
```

File: source/GameInterface/Utils/AutoSync/Dynamic/EventMessageGenerator.cs [hunk 23]
```diff
@@ -0,0 +1,120 @@
...
+
+        return constructor;
+    }
+
+    private void GenerateGetHashCode(TypeBuilder typeBuilder, FieldInfo[] fields)
+    {
+        MethodBuilder getHashCodeMethodBuilder = typeBuilder.DefineMethod("GetHashCode",
+                                            MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.NewSlot |
+                                            MethodAttributes.Virtual | MethodAttributes.Final,
+                                            typeof(int), null);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/EventMessageGenerator.cs:74]:
```suggestion
                                            typeof(int), Type.EmptyTypes);
```

File: source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs [hunk 24]
```diff
@@ -0,0 +1,125 @@
...
+
+        return constructor;
+    }
+
+    private void GenerateGetHashCode(TypeBuilder typeBuilder, FieldInfo[] fields)
+    {
+        MethodBuilder getHashCodeMethodBuilder = typeBuilder.DefineMethod("GetHashCode",
+                                            MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.NewSlot |
+                                            MethodAttributes.Virtual | MethodAttributes.Final,
+                                            typeof(int), null);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs]:
```suggestion
                                            typeof(int), Type.EmptyTypes);
```

File: source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs [hunk 25]
```diff
@@ -0,0 +1,151 @@
...
+
+        // Define a private field for the property
+        FieldBuilder fieldBuilder = typeBuilder.DefineField("_" + propertyName, propertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
+        fieldBuilder.SetCustomAttribute(customAttributeBuilder);
+
+        // Define the getter method
+        MethodBuilder getMethodBuilder = typeBuilder.DefineMethod("get_" + propertyName,
+                                        MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.SpecialName |
+                                        MethodAttributes.NewSlot | MethodAttributes.Virtual | MethodAttributes.Final,
+                                        propertyType, null);
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs:135]:
```suggestion
                                        propertyType, Type.EmptyTypes);
```

File: source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs [hunk 26]
```diff
@@ -0,0 +1,125 @@
...
+/// </summary>
+/// <remarks>
+/// An network message type is a type that implements <see cref="IAutoSyncMessage{T}"/> interface.
+/// Conains a single property of type <see cref="IAutoSyncData{T}"/>.
+/// </remarks>
+public class NetworkMessageGenerator
+{
+    public Type GenerateNetworkMessage(ModuleBuilder moduleBuilder, PropertyInfo property)
+    {
+        TypeBuilder typeBuilder = moduleBuilder.DefineType($"AutoSync_Network{property.DeclaringType.Name}_{property.Name}Changed",
```
Comment (samuelzurowski) [inline source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs]:
```suggestion
        TypeBuilder typeBuilder = moduleBuilder.DefineType($"AutoSync_Network{property.DeclaringType.Name}_{property.Name}_Changed",
```

Comment (samuelzurowski) [reply source/GameInterface/Utils/AutoSync/Dynamic/NetworkMessageGenerator.cs]: may not want the _ just a suggestion

**Review by samuelzurowski — CHANGES_REQUESTED** [review]
please check my comments

File: source/Common/Serialization/SerializableTypeMapper.cs (hunk 1 above)
Comment (samuelzurowski) [reply source/Common/Serialization/SerializableTypeMapper.cs:10]: sorry there's a class under it :) did this dumb stuff on a bunch of them.

**Comment by samuelzurowski** [comment]
@garrettluskey please check my comments

File: source/pdb/pingme.txt
Comment (garrettluskey) [reply source/pdb/pingme.txt:1]: I think that was created by DotPeek, removed

File: source/GameInterface/Services/MobileParties/Patches/PartyArmyPatches.cs (hunk 10 above)
Comment (garrettluskey) [reply source/GameInterface/Services/MobileParties/Patches/PartyArmyPatches.cs]: Yea removed

File: source/GameInterface/Utils/AutoSync/Template/AutoSyncHandlerTemplate.cs
Comment (garrettluskey) [reply source/GameInterface/Utils/AutoSync/Template/AutoSyncHandlerTemplate.cs:1]: This is not meant to be used outside of auto generation, updated name to reduce confusion

File: source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs (hunk 11 above)
Comment (garrettluskey) [reply source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs]: `Array.Empty<object>()` is actually faster

File: source/GameInterface.Tests/AutoSync/HandlerGeneratorTests.cs (hunk 15 above)
Comment (garrettluskey) [reply source/GameInterface.Tests/AutoSync/HandlerGeneratorTests.cs:69]: https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/Common/Messaging/MessageBroker.cs#L12

File: source/GameInterface.Tests/AutoSync/EventGeneratorTests.cs (hunk 14 above)
Comment (garrettluskey) [reply source/GameInterface.Tests/AutoSync/EventGeneratorTests.cs:38]: This is a .net core thing. Idk gets rid of the green squgglies. Really not needed but w/e

File: source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs (hunk 16 above)
Comment (garrettluskey) [reply source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs]: This actually inherits from an interface `IAutoSyncData`.
typeof(object) can be removed

File: source/GameInterface/Utils/AutoSync/Template/IAutoSyncData.cs (hunk 18 above)
Comment (garrettluskey) [reply source/GameInterface/Utils/AutoSync/Template/IAutoSyncData.cs:13]: https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/autosync/source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs#L39

Comment (garrettluskey) [reply source/GameInterface/Utils/AutoSync/Template/IAutoSyncData.cs:13]: The DataClassGenerator implements the fields of that interface

**Review by samuelzurowski — APPROVED** [review]

**Pull request merged**